kml.cc

```diff
@@ -1506,18 +1506,28 @@ void KmlFormat::kml_mt_simple_array(const route_head* header,
     case wp_field::igc_gfo:
     case wp_field::igc_siu:
     case wp_field::igc_acz: {
-      if (fs_igc->has_value(member)) {
-        double value = fs_igc->get_value(member).value();
-        if (global_opts.debug_level >= 6) {
-          printf(MYNAME ": Writing KML SimpleArray data: %s of %f\n", name, value);
+      if (fs_igc) {
+        if (fs_igc->has_value(member)) {
```

Comment:
I don't think we need to check fs_igc here.  I think has_igc_exts implies fs_igc != nullptr, and kml_mt_simple_array is only called with these members when has_igc_exts is true.  But it would be good to add an assertion:
assert(fs_igc != nullptr);
Assertions express the programmers intention and in non-debug code are compiled out.  In debug code they will stop execution with a message if violated.

If we have to check fs_igc then can't we collapse this to
if (fs_igc && fs_igc->has_value(member)) {
  ...
} else {
  ...
}